### ISSUE TITLE
How to share a folder to docker build context

### ISSUE THREAD

**Issue by nischal62019** [eclipse-che/che#1374]
Web
more
Forums
How to share a folder to docker build context

nischal reddy
to che-dev
1 hour agoDetails
Hi There,

I'm new to eclipse che and docker as well, trying to wrap my head around a problem.

I have installed eclipse che on windows with standard condenvy installer.

And i'm trying to copy a folder (on windows) which contains our product into docker container.

First i have copied the folder say "myFolder" i wanted to share into c:/Users since this is already mounted into the Virtual Box and is available in the Docker machine.

But COPY command is not working in the docker file and the docker context doesnt seem to have the folder "myFolder" in the build context.

How can i add "myFolder" to my docker build context so that i can use COPY command to copy the folder in to my docker container.

Any help would be much appreciated. 

We are working on a 3 day hackathon so sooner i get a reply works better for me :).

Thanks in advance,
Nischal Y

**Comment by ghost** [comment]
@nischal62019 Machine API reaches Docker directly, i.e. triggers Docker API through Unix socket or over tcp (depending on local settings). There's no build context as such (I think you expect it to work the way Docker CLI works).

As a workaround, instead of ADDing or COPYing your files, you may host them remotely and wget ot curl them.

**Comment by TylerJewell** [comment]
I was thinking of an alternative workaround.  Can files be volume mounted into the docker container into the /projects directory under a special name?

**Comment by nischal62019** [comment]
@eivantsov we indeed were thinking about hosting them remotely and getting them using curl/wget if COPY was not possible. But i'm bit worried since the size of the product is around 2gb.

@TylerJewell I did try volume mount with the following che property:

machine.server.extra.volume=/somepath/myFolder:/tmp/myFolder

but when i try 

RUN cp /tmp/myFolder /someotherFolder

I get error saying "/tmp/myFolder" is not available.

if i create my docker file without the cp statement the container is getting created and the volume is mounted properly. Correct me if i'm wrong, i assume volume is mounted once the container is up.

my folder has my product files which might change so i dont want to directly use it in the volume which might change the underlying files. That is the reason i'm trying to copy them into some other folder.

if copying from the mounted volume is not possible in dockerfile i will have to go with the approach suggested by  @eivantsov
